consensus/istanbul/backend/pos.go

```diff
@@ -91,6 +110,33 @@ func (sb *Backend) updateValidatorScores(header *types.Header, state *state.Stat
 	return nil
 }
 
+func (sb *Backend) getUptime(validatorIndex int, epoch uint64) (*big.Int, error) {
+	// try to get a handle on the core.blockchain's database
+	// sb.db is a different database (defined in CreateConsensusEngine) :/
+	// is there a better way than this? don't think we want to pollute the interface
+	c := sb.chain.(*core.BlockChain)
+	db := c.GetDatabase()
+	uptimes := rawdb.ReadAccumulatedEpochUptime(db, epoch)
+	if uptimes == nil {
+		panic(fmt.Sprintf("could not read uptimes, i: %v", validatorIndex))
+		return nil, errors.New("Invalid accumulated uptime")
+	}
+
+	// skip the math if we've counted more signatures than necessary for that epoch
+	// when does this happen?
+	if uptimes[validatorIndex].Score >= sb.EpochSize()-sb.LookbackWindow()+1 {
```

Comment:
This should never happen... should we return an error instead?

We should at the very least log an error

Reply:
function is removed